go/pkg/client/cas.go

```diff
@@ -1270,8 +1302,26 @@ func afterDownload(batch []digest.Digest, reqs map[digest.Digest][]*downloadRequ
 		if !ok {
 			log.Errorf("Precondition failed: download request not found in input %v.", dg)
 		}
-		for _, r := range rs {
-			r.wait <- err
+		lbMoved := dg.Size
+		rbMoved, ok := bytesMoved[dg]
+		// If there's no real bytes moved it likely means there was an error moving these.
+		if !ok {
+			lbMoved = 0
+		}
+		stats := MovedBytesMetadata{
+			Requested:    dg.Size,
+			LogicalMoved: lbMoved,
+			RealMoved:    rbMoved,
+		}
+		for i, r := range rs {
+			// bytesMoved will be zero for error cases.
+			// We only report it to the first client to prevent double accounting.
+			r.wait <- &downloadResponse{stats: stats, err: err}
+			if i == 0 {
+				stats.Cached = stats.LogicalMoved
+				stats.RealMoved = 0
+				stats.LogicalMoved = 0
```

Comment:
See the other comment about getting rid of "Cached" field. If you do that, then I think you may have to set LogicalMoved for the other clients as well.

Reply:
Fair point - but let's use the other comment to decide on keeping/not keeping Cached :)

Reply:
Resolving since we are going to be keeping Cached around.